src/general/world_effects/patch_events/MeteorImpactEvent.cs

```diff
@@ -0,0 +1,308 @@
+                return i;
+        }
+
+        throw new ArgumentException("Chances list is empty");
+    }
+
+    private void ChooseAffectedPatches()
+    {
+        var impactSize = random.Next(0, 3);
+        var surfacePatches = targetWorld.Map.Patches.Values.Where(IsSurfacePatch).ToList();
+        var selectedPatch = surfacePatches[random.Next(surfacePatches.Count)];
+        var adjacentRegion = selectedPatch.Region.Adjacent.ToList()[random.Next(selectedPatch.Region.Adjacent.Count)];
+
+        // 1 patch
+        if (impactSize >= 0)
+        {
+            modifiedPatchesIds.Add(selectedPatch.ID);
+        }
+
+        // all surface patches in region
+        if (impactSize >= 1)
+        {
+            foreach (var adjacent in selectedPatch.Adjacent)
+            {
+                if (adjacent.Region.ID == selectedPatch.Region.ID && IsSurfacePatch(adjacent))
+                {
+                    modifiedPatchesIds.Add(adjacent.ID);
+                }
+            }
+        }
+
+        // all surface patches in 2 neighbouring regions
+        if (impactSize >= 2)
+        {
+            foreach (var patch in adjacentRegion.Patches)
+            {
+                if (IsSurfacePatch(patch))
+                {
+                    modifiedPatchesIds.Add(patch.ID);
+                }
+            }
+        }
+    }
+
+    private bool AreConditionsMet()
+    {
+        return random.NextFloat() <= Constants.METEOR_IMPACT_CHANCE;
+    }
+
+    private void ChangePatchProperties(Patch patch, double totalTimePassed)
+    {
+        AdjustEnvironmentalConditions(patch);
+        AdjustCompounds(patch);
+        AddChunks(patch);
+        LogEvent(patch, totalTimePassed);
+    }
+
+    /// <summary>
+    ///   Gets chunks from event template patch and applies them to the patches.
+    /// </summary>
+    private void AddChunks(Patch patch)
+    {
+        var templateBiome = SimulationParameters.Instance.GetBiome(TemplateBiomeForChunks);
+        foreach (var configuration in selectedMeteor.Chunks)
+        {
+            var configurationExists =
+                templateBiome.Conditions.Chunks.TryGetValue(configuration, out var chunkConfiguration);
+            if (!configurationExists)
+            {
+                GD.PrintErr("Event chunk configuration does not exist: " + configuration);
+                continue;
+            }
+
+            chunkConfiguration.Density *= random.NextFloat() * 0.5f + 0.5f;
+            patch.Biome.Chunks.Add(configuration, chunkConfiguration);
+        }
+    }
+
+    private void AdjustEnvironmentalConditions(Patch patch)
+    {
+        bool hasSunlight = patch.Biome.ChangeableCompounds.TryGetValue(Compound.Sunlight, out var currentSunlight);
+
+        if (!hasSunlight)
+        {
+            GD.PrintErr("Meteor impact event encountered patch with unexpectedly no sunlight");
+            return;
+        }
+
+        currentSunlight.Ambient *= Constants.METEOR_IMPACT_SUNLIGHT_MULTIPLICATION;
```

Comment:
I see this is how you handle multiple events overlapping. I suppose mathwise this does check out.

But is the gameplay impact of for example glaciation and a meteor impact good at the same time? Reducing sunlight even below 50% is probably very punishing.